### ISSUE TITLE
Did you predict the left and right disparity by only the left image?

### ISSUE THREAD

**Issue by leonzfa** [mrharicot/monodepth#32]
In the "monodepth_model.py", if "do_stereo" is false, the input for the network will be only the left image:

if self.params.do_stereo:
       self.model_input = tf.concat([self.left, self.right], 3)
else:
       self.model_input = self.left

But the output of the network contains both left disparity and right disparity:

def get_disp(self, x):
       disp = 0.3 * self.conv(x, 2, 3, 1, tf.nn.sigmoid)
       return disp

Is this the case in your code? If so, why not used the right image to predict the right disparity?

Thank you very much.

**Comment by mrharicot** [comment]
Hi,

This is the idea behind the method, we do single frame depth estimation, but estimate both disparities and make them consistent with each other.
Check our paper, we explain it in details.

**Comment by leonzfa** [comment]
Thanks a lot
